backend/escalation_engine.py

```
@@ -0,0 +1,286 @@
+        """
+        now = datetime.datetime.now(datetime.timezone.utc)
+
+        # Get grievances that are active and past SLA deadline
+        return db.query(Grievance).filter(
+            and_(
+                Grievance.status.in_([GrievanceStatus.OPEN, GrievanceStatus.IN_PROGRESS, GrievanceStatus.ESCALATED]),
+                Grievance.sla_deadline < now
+            )
+        ).all()
+
+    def _should_escalate(self, grievance: Grievance, db: Session) -> bool:
```

Ruff (0.14.14)

[warning] Unused method argument: `db`

(ARG002)